core/src/test/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceServiceUnitTestsSpec.scala

```diff
@@ -0,0 +1,307 @@
+
+    doReturn(Future()).when(service).requireCreateWorkspaceAction(billingProject.projectName, requestContext)
+
+    val billingProfile = mock[ProfileModel]
+    when(billingProfile.getCloudPlatform()).thenReturn(CloudPlatform.AZURE)
+    doReturn(Future(Some(billingProfile))).when(service).getBillingProfile(billingProject, requestContext)
+
+    val destWorkspace = mock[Workspace]
+    doReturn(Future(destWorkspace))
+      .when(service)
+      .cloneAzureWorkspaceAsync(
+        ArgumentMatchers.eq(sourceWorkspace),
+        ArgumentMatchers.eq(billingProfile),
+        ArgumentMatchers.eq(destWorkspaceRequest),
+        ArgumentMatchers.any()
+      )
+
+    whenReady(
+      service.cloneMultiCloudWorkspaceAsync(
+        mock[WorkspaceService],
+        sourceWorkspace.toWorkspaceName,
+        destWorkspaceRequest
+      )
+    )(_ shouldBe destWorkspace)
+
+  }
+
+  behavior of "cloneAzureWorkspaceAsync"
+
+  it should "delete the new workspace on failures" in {
+    val sourceWorkspaceName = "source-name"
+    val sourceWorkspaceNamespace = "source-namespace"
+    val sourceWorkspace = Workspace.buildMcWorkspace(
+      sourceWorkspaceNamespace,
+      sourceWorkspaceName,
+      UUID.randomUUID().toString,
+      DateTime.now(),
+      DateTime.now(),
+      "creator",
+      Map(),
+      WorkspaceState.Ready
+    )
+    val destWorkspaceRequest = WorkspaceRequest("dest-namespace", "dest-name", Map())
+
+    val requestContext = mock[RawlsRequestContext]
+    when(requestContext.otelContext).thenReturn(None)
+
+    val workspaceManagerDAO = mock[WorkspaceManagerDAO]
+    doAnswer(_ => throw new ApiException())
+      .when(workspaceManagerDAO)
+      .cloneWorkspace(any(), any(), any(), any(), any(), any(), any())
+
+    val service = spy(
+      MultiCloudWorkspaceService.constructor(
+        mock[SlickDataSource],
+        workspaceManagerDAO,
+        mock[BillingProfileManagerDAO],
+        mock[SamDAO],
+        mock[MultiCloudWorkspaceConfig],
+        mock[LeonardoDAO],
+        "MultiCloudWorkspaceService-test"
+      )(requestContext)
+    )
+
+    val billingProfile = mock[ProfileModel]
+    when(billingProfile.getCreatedDate).thenReturn(DateTime.now().toString)
+
+    val destWorkspace = mock[Workspace]
+    doReturn(Future(destWorkspace))
+      .when(service)
+      .createNewWorkspaceRecord(
+        ArgumentMatchers.any(),
+        ArgumentMatchers.eq(destWorkspaceRequest),
+        ArgumentMatchers.eq(requestContext),
+        ArgumentMatchers.eq(WorkspaceState.Cloning)
+      )
+    doReturn(Future(true)).when(service).deleteWorkspaceRecord(destWorkspace)
+
+    intercept[ApiException] {
+      Await.result(
+        service.cloneAzureWorkspaceAsync(sourceWorkspace, billingProfile, destWorkspaceRequest, requestContext),
+        Duration.Inf
+      )
+    }
+
+    verify(service).deleteWorkspaceRecord(destWorkspace)
+  }
+
+  it should "doesn't try to delete the workspace when the new workspace creation fails" in {
```

Comment:
The failure is specifically in creating the Rawls workspace record?

Reply:
yes
there's another test that checks that it _does_ delete the record otherwise

Reply:
Can you make the test description clearer? Because it SHOULD try to delete the workspace other parts of workspace creation fail.